pkg/scheduler/partition.go

```diff
@@ -1222,3 +1240,12 @@ func (pc *PartitionContext) GetNodeSortingPolicy() policies.SortingPolicy {
 	defer pc.RUnlock()
 	return pc.nodeSortingPolicy.PolicyType
 }
+
+func (pc *PartitionContext) moveCompletedApp(appID string) {
+	pc.Lock()
+	defer pc.Unlock()
+	if app, ok := pc.applications[appID]; ok {
+		delete(pc.applications, appID)
+		pc.completedApplications[appID] = app
```

Comment:
We need to be careful: there is no reason to assume that the same application ID is not re-used multiple times in the time we keep the old application around. A simple time stamp addition will allow us to re-use the appID multiple times:
```
// new ID as completedApplications map key, use negative value to get a divider
newID := appID + strconv.FormatInt(-(time.Now()).Unix(), 10)
```
The `newID` is then only used as the key in the _completedApplications_ map. No updates in the application object.